tests/lib/rules/no-array-prototype-extensions.js

```diff
@@ -740,10 +740,85 @@ import { compare as comp } from '@custom/utils';
       errors: [{ messageId: 'main', type: 'CallExpression' }],
     },
     {
+      // When unexpected number of params are passed, we will skip auto-fixing
       code: 'something.uniqBy()',
       output: null,
       errors: [{ messageId: 'main', type: 'CallExpression' }],
     },
+    {
+      // Primitive value as an argument
+      code: 'something.uniqBy(1)',
+      output: `import { get } from '@ember/object';
+something.reduce(([uniqArr, itemsSet, getterFn], item) => {
+          const val = getterFn(item);
+          if (!itemsSet.has(val)) {
+            itemsSet.add(val);
+            uniqArr.push(item);
+          }
+          return [uniqArr, itemsSet, getterFn];
+        }, [[], new Set(), typeof 1 === 'function' ? 1 : (item) => get(item, 1)])[0]`,
```

Comment:
Checking `typeof 1 === 'function'` seems non-sensical, can we evaluate that ourselves and avoid the ternary?

Reply:
Makes sense. Addressed the comment.